### ISSUE TITLE
UI accessed from non-UI thread

### ISSUE THREAD

**Issue by mrward** [mono/monodevelop#4007]
With Visual Studio 2017 for Mac (Preview) 7.5 Preview (7.5 build 654) (master build from Wrench), saw this entry in the IDE log:

```
AppKit.AppKitThreadAccessException: AppKit Consistency error: you are calling a method that can only be invoked from the UI thread.
  at AppKit.NSApplication.EnsureUIThread () [0x00020] in /Library/Frameworks/Xamarin.Mac.framework/Versions/4.5.0.18/src/Xamarin.Mac/AppKit/NSApplication.cs:107 
  at AppKit.NSAccessibilityElement.set_AccessibilityRole (System.String value) [0x00001] in /Library/Frameworks/Xamarin.Mac.framework/Versions/4.5.0.18/src/Xamarin.Mac/AppKit/NSAccessibilityElement.g.cs:3558 
  at MonoDevelop.Components.AtkCocoaHelper.RealAccessibilityElementProxy.SetRole (System.String role, System.String description) [0x00000] in /Users/builder/data/lanes/432/e3d445bc/source/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Components.AtkCocoaHelper/AtkCocoaHelperMac.cs:1133 
  at MonoDevelop.Components.AtkCocoaHelper.RealAccessibilityElementProxy.SetRole (MonoDevelop.Components.AtkCocoaHelper.AtkCocoa+Roles role, System.String description) [0x00000] in /Users/builder/data/lanes/432/e3d445bc/source/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Components.AtkCocoaHelper/AtkCocoaHelperMac.cs:1141 
  at MonoDevelop.Components.AtkCocoaHelper.AccessibilityElementProxy.SetRole (MonoDevelop.Components.AtkCocoaHelper.AtkCocoa+Roles role, System.String description) [0x00000] in /Users/builder/data/lanes/432/e3d445bc/source/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Components.AtkCocoaHelper/AtkCocoaHelperMac.cs:685 
  at MonoDevelop.Components.PathEntry.get_Accessible () [0x0001a] in /Users/builder/data/lanes/432/e3d445bc/source/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Components/PathBar.cs:81 
  at MonoDevelop.Components.PathBar.UpdatePathAccessibility () [0x00031] in /Users/builder/data/lanes/432/e3d445bc/source/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Components/PathBar.cs:233 
  at MonoDevelop.Components.PathBar.SetPath (MonoDevelop.Components.PathEntry[] path) [0x000b1] in /Users/builder/data/lanes/432/e3d445bc/source/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Components/PathBar.cs:276 
  at MonoDevelop.Ide.Gui.SdiWorkspaceWindow.HandlePathChange (System.Object sender, MonoDevelop.Ide.Gui.Content.DocumentPathChangedEventArgs args) [0x0000f] in /Users/builder/data/lanes/432/e3d445bc/source/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/SdiWorkspaceWindow.cs:725 
  at (wrapper delegate-invoke) System.EventHandler`1[MonoDevelop.Ide.Gui.Content.DocumentPathChangedEventArgs].invoke_void_object_TEventArgs(object,MonoDevelop.Ide.Gui.Content.DocumentPathChangedEventArgs)
  at MonoDevelop.CSharp.PathedDocumentTextEditorExtension.OnPathChanged (MonoDevelop.Ide.Gui.Content.DocumentPathChangedEventArgs e) [0x0000a] in /Users/builder/data/lanes/432/e3d445bc/source/monodevelop/main/src/addins/CSharpBinding/MonoDevelop.CSharp/PathedDocumentTextEditorExtension.cs:361 
  at MonoDevelop.CSharp.PathedDocumentTextEditorExtension+<>c__DisplayClass52_0+<<Update>b__0>d.MoveNext () [0x0011e] in /Users/builder/data/lanes/432/e3d445bc/source/monodevelop/main/src/addins/CSharpBinding/MonoDevelop.CSharp/PathedDocumentTextEditorExtension.cs:752 
```

Full callstack:

https://gist.github.com/mrward/b87d166fe199eb9d4d69145735d56200

Looking at the code it is doing a Task.Run and then firing the OnPathChanged event. So either the event should be fired on the UI thread, or everybody who uses that event needs to run it on the UI thread themselves.

https://github.com/mono/monodevelop/blob/master/main/src/addins/CSharpBinding/MonoDevelop.CSharp/PathedDocumentTextEditorExtension.cs#L742-L752

> VS bug #574358

**Comment by xamarin-release-manager** [comment]
Fixed in version 7.6.0.3 (master)

Pull Request #4017 merged by: Mike Krüger
Author: mono
Commit: d4c1e962fdd9df2fe2acd3b4bf008c93915b346f (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 7.5.0.757 (d15-7)

Author: Mike Krüger
Commit: 44edbb10bfafcc16d78cb9a414fabbaf11d03784 (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.6.0.802 (master)

Pull Request #4017 merged by: Mike Kr??ger
Author: mono
Commit: d4c1e962fdd9df2fe2acd3b4bf008c93915b346f (xamarin/monodevelop)

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-16] Comment by **xamarinc**



Fixed in version 8.6.0.802 (master)



Pull Request #4017 merged by: Mike Kr??ger  
 Author: mono  
 Commit: d4c1e962fdd9df2fe2acd3b4bf008c93915b346f (xamarin/monodevelop)
